modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/ldap/LdapAuthHandler.java

```diff
@@ -0,0 +1,158 @@
+                    .toAbsolutePath().toString();
+            try (FileInputStream fin = new FileInputStream(trustStoreLocation)) {
+
+                KeyStore trustStore = KeyStore.getInstance(ldapConfiguration.getSsl().getTrustStore().getType());
+                trustStore.load(fin, ldapConfiguration.getSsl().getTrustStore().getPassword().toCharArray());
+                TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(
+                        ldapConfiguration.getSsl().getTrustStore().getCertType());
+                trustManagerFactory.init(trustStore);
+                SSLContext sslContext = SSLContext.getInstance(ldapConfiguration.getSsl().getProtocol());
+                sslContext.init(null, trustManagerFactory.getTrustManagers(), null);
+                LdapSslSocketFactory.setSslContext(sslContext);
+
+            } catch (KeyStoreException | IOException | NoSuchAlgorithmException | CertificateException
+                    | KeyManagementException e) {
+                throw new AuthException("Initializing ldap trust store failed", e);
+            }
+        }
+    }
+
+    public DirContext connectAnonymously() throws NamingException {
+
+        return connect(new Hashtable<String, String>());
+    }
+
+    public DirContext connect(Hashtable<String, String> env) throws NamingException {
+
+        DirContext dirContext = null;
+        env.put(Context.INITIAL_CONTEXT_FACTORY, LdapConstants.JNDI_LDAP_CTX_FACTORY);
+        env.put(Context.SECURITY_AUTHENTICATION, LdapConstants.LDAP_AUTHENTICATION_SIMPLE);
+
+        if (ldapConfiguration.getSsl().isEnabled()) {
+
+            env.put(Context.PROVIDER_URL,
+                    "ldap://" + ldapConfiguration.getHostName() + ":" + ldapConfiguration.getSsl().getPort());
+            env.put(Context.SECURITY_PROTOCOL, LdapConstants.SECURITY_PROTOCOL_SSL);
+            env.put(LdapConstants.ENV_KEY_LDAP_SOCKET_FACTORY, LdapSslSocketFactory.class.getName());
+            dirContext = new InitialDirContext(env);
+
+        } else {
+
+            env.put(Context.PROVIDER_URL,
+                    "ldap://" + ldapConfiguration.getHostName() + ":" + ldapConfiguration.getPlain().getPort());
+            dirContext = new InitialDirContext(env);
+        }
+
+        return dirContext;
+    }
+
+    public String searchDN(DirContext dirContext, String username) throws NamingException {
+        String lookup = ldapConfiguration.getUsernameSearchFilter().replace("?", username);
+        SearchControls ctrl = new SearchControls();
+        ctrl.setSearchScope(SearchControls.SUBTREE_SCOPE);
+
+        NamingEnumeration<SearchResult> answer = dirContext.search(ldapConfiguration.getBaseDN(), lookup, ctrl);
+
+        String dn;
+        if (answer.hasMore()) {
+            javax.naming.directory.SearchResult result = answer.next();
+            dn = result.getNameInNamespace();
+            LOGGER.debug("DN: " + dn + " obtained for Username: " + username);
```

Comment:
Use the built-in formatting to construct this argument. Let's fix in other places as well